File: app/src/main/java/com/niiadotei/storehouse/ui/suppliers/SuppliersViewModel.java
package com.niiadotei.storehouse.ui.suppliers;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.niiadotei.storehouse.R;
import com.niiadotei.storehouse.data.DatabaseHelper;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class SuppliersViewModel extends RecyclerView.Adapter<SuppliersViewModel.ViewHolder> {
    AppCompatActivity appCompatActivity;

    JSONArray jsonArray;

    DatabaseHelper databaseHelper;

    public SuppliersViewModel(AppCompatActivity appCompatActivity, JSONArray jsonArray) {
        this.appCompatActivity = appCompatActivity;
        this.jsonArray = jsonArray;
    }

    @NonNull
    @Override
    public SuppliersViewModel.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_suppliers, parent, false);

        databaseHelper = new DatabaseHelper(view.getContext());

        return  new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SuppliersViewModel.ViewHolder holder, int position) {
        try {
            JSONObject jsonObject = jsonArray.getJSONObject(position);

            holder.supplierNameTextView.setText(jsonObject.getString("name"));
            holder.supplierPhoneNumberTextView.setText(jsonObject.getString("phone"));
            holder.supplierLocationTextView.setText(jsonObject.getString("location"));
            holder.supplierDisplayNameTextView.setText(jsonObject.getString("display"));
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    @Override
    public int getItemCount() {
        return jsonArray.length();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView supplierNameTextView;
        TextView supplierPhoneNumberTextView;
        TextView supplierLocationTextView;
        TextView supplierDisplayNameTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            supplierNameTextView = itemView.findViewById(R.id.supplierNameTextView);
            supplierPhoneNumberTextView = itemView.findViewById(R.id.supplierPhoneNumberTextView);
            supplierLocationTextView = itemView.findViewById(R.id.supplierLocationTextView);
            supplierDisplayNameTextView = itemView.findViewById(R.id.supplierDisplayNameTextView);
        }
    }
}

File: app/src/main/java/com/niiadotei/storehouse/ui/suppliers/SuppliersViewModel.kt
package com.niiadotei.storehouse.ui.suppliers

import org.json.JSONArray
import androidx.recyclerview.widget.RecyclerView
import com.niiadotei.storehouse.data.DatabaseHelper
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import com.niiadotei.storehouse.R
import org.json.JSONException
import android.widget.TextView

class SuppliersViewModel(var jsonArray: JSONArray) : RecyclerView.Adapter<SuppliersViewModel.ViewHolder>() {
    private lateinit var databaseHelper: DatabaseHelper

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_suppliers, parent, false)
        databaseHelper = DatabaseHelper(view.context)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        try {
            val jsonObject = jsonArray.getJSONObject(position)

            holder.supplierNameTextView.text = jsonObject.getString("name")

            val phone = "Contact: " + jsonObject.getString("phone")
            holder.supplierPhoneNumberTextView.text = phone

            val location = "Location: " + jsonObject.getString("location")
            holder.supplierLocationTextView.text = location

            holder.supplierDisplayNameTextView.text = jsonObject.getString("display")
        } catch (e: JSONException) {
            e.printStackTrace()
        }
    }

    override fun getItemCount(): Int {
        return jsonArray.length()
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var supplierNameTextView: TextView
        var supplierPhoneNumberTextView: TextView
        var supplierLocationTextView: TextView
        var supplierDisplayNameTextView: TextView

        init {
            supplierNameTextView = itemView.findViewById(R.id.supplierNameTextView)
            supplierPhoneNumberTextView = itemView.findViewById(R.id.supplierPhoneNumberTextView)
            supplierLocationTextView = itemView.findViewById(R.id.supplierLocationTextView)
            supplierDisplayNameTextView = itemView.findViewById(R.id.supplierDisplayNameTextView)
        }
    }
}